recipes/llama_api_providers/examples_with_aws/Prompt_Engineering_with_Llama_3_On_Amazon_Bedrock.ipynb

```diff
@@ -0,0 +1,2358 @@
+    "# listModels = bedrock.list_foundation_models(byProvider='meta')\n",
+    "# print(\"\\n\".join(list(map(lambda x: f\"{x['modelName']} : { x['modelId'] }\", listModels['modelSummaries']))))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 315,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "LLAMA3_70B_INSTRUCT = \"meta.llama3-70b-instruct-v1:0\"\n",
+    "LLAMA3_8B_INSTRUCT = \"meta.llama3-8b-instruct-v1:0\"\n",
+    "# We'll default to the smaller 8B model for speed; change to LLAMA3_70B_CHAT for more advanced (but slower) generations\n",
+    "DEFAULT_MODEL = LLAMA3_8B_INSTRUCT\n",
+    "\n",
+    "\n",
+    "def completion(\n",
+    "    prompt: str,\n",
+    "    model: str = DEFAULT_MODEL,\n",
+    "    temperature: float = 0.6,\n",
+    "    top_p: float = 0.9,\n",
+    ") -> str:\n",
+    "    llm = Bedrock(model_id=model)\n",
+    "    prompt_str = ''.join(prompt)\n",
+    "    if not prompt_str.endswith(\"<|eot_id|><|start_header_id|>assistant<|end_header_id|>\"):\n",
+    "        prompt += \"<|eot_id|><|start_header_id|>assistant<|end_header_id|>\"\n",
+    "    return llm.invoke(prompt, temperature=temperature, top_p=top_p)\n",
+    "\n",
+    "\n",
+    "def chat_completion(\n",
+    "    messages: List[Dict],\n",
+    "    model=DEFAULT_MODEL,\n",
+    "    temperature: float = 0.6, \n",
+    "    top_p: float = 0.9,\n",
+    ") -> str:\n",
+    "    prompt_str = ''.join([msg['content'] for msg in messages])\n",
+    "    return completion(\n",
+    "        prompt_str,\n",
+    "        model,\n",
+    "        temperature,\n",
+    "        top_p,\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "def assistant(content: str):\n",
+    "    return {\"role\": \"assistant\", \"content\": content}\n",
+    "\n",
+    "\n",
+    "def user(content: str):\n",
+    "    return {\"role\": \"user\", \"content\": content}\n",
+    "\n",
+    "\n",
+    "def complete_and_print(prompt: str, model: str = DEFAULT_MODEL, verbose: bool = True):\n",
+    "    print(f'==============\\n{prompt}\\n==============')\n",
+    "    if not prompt.endswith(\"<|eot_id|><|start_header_id|>assistant<|end_header_id|>\"):\n",
+    "        prompt += \"<|eot_id|><|start_header_id|>assistant<|end_header_id|>\"\n",
+    "    response = completion(prompt, model)\n",
+    "    if verbose: \n",
+    "        print(response, end='\\n\\n')\n",
+    "    return response\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Completion APIs\n",
+    "\n",
+    "Llama 2 models tend to be wordy and explain their rationale. Later we'll explore how to manage the response length."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 316,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "==============\n",
```

Comment:
I see you have saved the output of these prompts across the notebook. Ignore my first comment if thats how are you are creating all the notebooks. If its not a standard lets create this output as well.